docs/source/articles/implement-the-credit-card-tender-type.html.md

```diff
@@ -0,0 +1,310 @@
+In either case, you can complete the credit card tender type implementation in the following steps:
+
+1. [Implement and configure credit card gateway(s)](#gateways_1)
+2. [Configure proxy (Commerce Cloud only)](#proxy_2)
+3. [Add test support](#test-support_3)
+4. [Implement credit card tokenization](#tokenization_4)
+5. [Implement each credit card payment operation](#operation-implementations_5)
+
+The following sections cover each step in greater depth and provide examples from
+_Workarea Core_ v3.4.18 (
+[gem](https://rubygems.org/gems/workarea-core/versions/3.4.18),
+[docs](https://www.rubydoc.info/gems/workarea-core/3.4.18),
+[source](https://github.com/workarea-commerce/workarea/tree/v3.4.18/core)
+)
+and
+_Workarea Braintree_ v1.0.3 (
+[gem](https://rubygems.org/gems/workarea-braintree/versions/1.0.3),
+[docs](https://www.rubydoc.info/gems/workarea-braintree/1.0.3),
+[source](https://github.com/workarea-commerce/workarea-braintree/tree/v1.0.3)
+).
+
+
+Gateways
+----------------------------------------------------------------------
+
+Workarea [configures](/articles/configuration.html) the credit card gateway as an instance of
+[`ActiveMerchant::Billing::BogusGateway`](https://www.rubydoc.info/gems/activemerchant/1.99.0/ActiveMerchant/Billing/BogusGateway), as shown below
+([source](https://github.com/workarea-commerce/workarea/blob/v3.4.18/core/config/initializers/11_payment.rb#L6)):
+
+```ruby
+if Workarea.config.gateways.credit_card.blank?
+  Workarea.config.gateways.credit_card =
+    ActiveMerchant::Billing::BogusGateway.new
+end
+```
+
+This gateway, which is the default bogus gateway from Active Merchant, is useful for local development and testing.
+However, for production use, you must choose or implement a gateway that connects to an actual payment service provider, and you must configure Workarea to use this gateway.
+You may also want to change the gateway(s) used in non-production environments.
+
+__Workarea requires credit card gateways to support tokenization.__
+( See section [Tokenization](#tokenization_4), below.)
+
+Workarea Braintree, a plugin that integrates Workarea with [Braintree](https://www.braintreepayments.com/), relies on
+[`ActiveMerchant::Billing::BraintreeGateway`](https://www.rubydoc.info/gems/activemerchant/1.99.0/ActiveMerchant/Billing/BraintreeGateway),
+from Active Merchant, as the production gateway.
+The plugin also extends Active Merchant to provide its own bogus gateway, 
+[`ActiveMerchant::Billing::BogusBraintreeGateway`](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/lib/active_merchant/billing/bogus_braintree_gateway.rb).
+
+Workarea Braintree provides code to autoconfigure Workarea's credit card gateway.
+When Braintree credentials are present, the Braintree gateway is used, otherwise the bogus gateway is used
+([source](https://github.com/workarea-commerce/workarea-braintree/blob/v1.0.3/lib/workarea/braintree.rb#L8-L24)):
+
+```ruby
+def self.auto_configure_gateway
```

Comment:
I'm not a fan of including this anti-pattern in our documentation, even though it's currently how credit card gateways are implemented. The issue is that auto-configuring the gateway in an initializer doesn't work in multi-site if each site has different credentials (which is totally possible in an international setting). Either we need to draw a line in the sand and drop support for per-site credentials out-of-box, or we need to change the way that credit card gateways are instantiated so they can occur per-request, after the `Site.current` has been set. But we can't throw this out there and pretend like it works all the time, because SIs and clients constantly have trouble with this whole concept, especially when it gets down to the wire of launch date...

Reply:
The Workarea team agreed this pattern should be updated in all the plugins that use it. We're going to do that work in a future sprint, after which I'll update this section with the updated examples.